src/main/java/com/cloudbees/groovy/cps/tool/Translator.java

```diff
@@ -246,26 +247,30 @@ public JExpression visitWhileLoop(WhileLoopTree wt, Void __) {
             @Override
             public JExpression visitMethodInvocation(MethodInvocationTree mt, Void __) {
                 ExpressionTree ms = mt.getMethodSelect();
-                JInvocation inv = $b.invoke("functionCall")
-                        .arg(loc(mt));
+                JInvocation inv;
 
                 if (ms instanceof MemberSelectTree) {
                     MemberSelectTree mst = (MemberSelectTree) ms;
-                    inv
+                    inv = $b.invoke("functionCall")
+                        .arg(loc(mt))
                         .arg(visit(mst.getExpression()))
                         .arg(n(mst.getIdentifier()));
                 } else
                 if (ms instanceof JCIdent) {
                     // invocation without object selection, like  foo(bar,zot)
                     JCIdent it = (JCIdent) ms;
-                    if (!it.sym.owner.toString().equals(DefaultGroovyMethods.class.getName())) {
+                    if (!it.sym.owner.toString().equals(fqcn)) {
                         // static import
-                        inv.arg($b.invoke("constant").arg(t(it.sym.owner.type).dotclass()))
-                           .arg(n(it));
+                        inv = $b.invoke("functionCall")
+                            .arg(loc(mt))
+                            .arg($b.invoke("constant").arg(t(it.sym.owner.type).dotclass()))
+                            .arg(n(it));
                     } else {
                         // invocation on this class
-                        inv.arg($b.invoke("this_"))
```

Comment:
Was translating

```java
each((Iterable) this, closure);
```

to an instance call; since there was no receiver, `this` was null and was throwing an exception.